src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,87 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String CSV_FILE_SEPERATOR = ",";
+    private static final String OPER_TYPE_SUPPLY = "supply";
+    private static final String OPER_TYPE_BUY = "buy";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String dataFromFile = readDataFromFile(fromFileName);
+
+        String report = generateReportFromData(dataFromFile);
+
+        writeDataToFile(report, toFileName);
+    }
+
+    private String readDataFromFile(String fromFileName) {
+        StringBuilder stringBuilder = new StringBuilder();
+
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+
+            String lineFromFile = bufferedReader.readLine();
+            while (lineFromFile != null) {
+                stringBuilder.append(lineFromFile).append(System.lineSeparator());
+
+                lineFromFile = bufferedReader.readLine();
+            }
+
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException("File " + fromFileName + " not found!", e);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file: " + fromFileName, e);
+        }
+
+        return stringBuilder.toString();
+    }
+
+    private String generateReportFromData(String data) {
```

Comment:
The `generateReportFromData` method is quite long and does several things: it splits the data into records, processes each record, and generates a report. It would be better to split this method into several smaller methods, each with a single responsibility. This will make the code easier to read and maintain.